app/javascript/src/editable_components.js

```diff
@@ -761,38 +766,24 @@ class EditableCollectionItemInjector {
   }
 }
 
-
-class EditableCollectionItemRemover {
-  constructor(editableCollectionItem, editableCollectionFieldComponent, config) {
-    var conf = mergeObjects({}, config);
-    var text = mergeObjects({ itemRemove: 'remove' }, config.text);
-    var $node = $(createElement("button", text.itemRemove, conf.classes));
-
-    $node.data("instance", this);
-    $node.addClass("EditableCollectionItemRemover");
-    $node.attr("type", "button");
-    $node.on("click.EditableCollectionItemRemover", function(e) {
-      e.preventDefault();
-      editableCollectionItem.remove();
-    });
-
-    // Close on ENTER || SPACE
-    $node.on("keydown.EditableCollectionItemRemover", function(e) {
-      e.preventDefault();
-      if(e.which == 13 || e.which == 32) {
-        editableCollectionItem.remove();
+function createEditableCollectionItemMenu(item, config) { 
+  var template = $("[data-component-template=EditableCollectionItemMenu]");
+  var $ul = $(template.html());
+
+  item.$node.append($ul);
+  
+  let menu = new EditableCollectionItemMenu($ul, {
+      activator_text: config.text.edit,
+      container_id: uniqueString("activatedMenu-"),
+      collectionItem: item,
+      menu: {
+        position: { my: "left top", at: "right-15 bottom-15" } // Position second-level menu in relation to first.
       }
     });
 
-    editableCollectionItem.$node.append($node);
-
-    this.component = editableCollectionFieldComponent;
-    this.item = editableCollectionItem;
-    this.$node = $node;
-  }
+    item.$node.data("ActivatedMenu", menu);
```

Comment:
Think I saw something similar to this line being removed somewhere in the old code, so guessing it's just based on what was found. 

Could do with a tweak if staying in this form to be more clear.
Assuming I'm understanding it correctly, the line itself should be updated to better illustrate the correct type of object being stored (guessing it was previously an ActivatedMenu).
`item.$node.data("EditableCollectionItemMenu", menu);`

However, there's another opportunity to improve on old code here, since it should be following some patterns we apply consistently elsewhere. 

Instead of this:
`item.$node.data("EditableCollectionItemMenu", menu);`

it should be something like this:
`item.menu = menu`

That's less jQuery-ish and more OOP. 

In fact, going with that approach, perhaps we could just return the `menu` instance at end of function so that the calling constructor could have something like this line:
`this.menu = createEditableCollectionItemMenu(this, config);`

Reply:
Yep, I just relocated a line and modified it.  Went with your last suggestion, return the menu form the function and assign it to the variable in the constructor. 👍